File: stochasticSearch/stochastic.js



function obj_function(vector) {
  return vector.reduce(function(prev, cur, i, arr) {
    return cur + (i * 2);
  });
}

function random_vector(minmax) {
  var arr = [];

}

(function() {

  var search_space = new Array(10);

  for (var i = 0; i < 10; i++) {
    search_space[i] = new Array(10);
  }

  console.log(search_space);

  for (var i = 0; i < search_space.length; i++) {
    for (var j = 0; j < search_space.length; j++) {
      search_space[i][j] = Math.random() * 10;
    }
  }
  console.log(search_space);
})();

File: stochasticSearch/stochastic.rb


#Stochastic search

def obj_function(vector)
  return vector.inject(0) {|sum, x| sum + (x ** 2.0)}
end

def random_vector(minmax)
  return Array.new(minmax.size) do |i|
    minmax[i][0] + ((minmax[i][1] - minmax[i][0]) * rand())
  end
end

def search(search_space, max_iter)
  best = nil
  max_iter.times do |iter|
    candidate = {}
    candidate[:vector] = random_vector(search_space)
    candidate[:cost] = obj_function(candidate[:vector])
    best = candidate if best.nil? or candidate[:cost] < best[:cost]
    puts " > iteration = #{(iter+1)}, best = #{best[:cost]}"
  end
  return best
end

if __FILE__ == $0
  problem_size = 2
  search_space = Array.new(problem_size) {|i| [-5, +5]}
  max_iter = 100000
  #best = search(search_space, max_iter)
  #puts "Done. Best Solution: c=#{best[:cost]}, v=#{best[:vector].inspect}"
  puts "Search space: #{search_space}"
end
